### DISCUSSION TITLE
How to use only half of the ColorChecker Passport?

### DISCUSSION THREAD

**Discussion by YiteLiu** [colour-science/colour-checker-detection#31] [Q&A] [answered]
Hi, thanks for your contribution.  I am a newbie for coding or GitHub, hope that question does not bother you. I did not set my color checker like your example. I want to use only half of the x-rite color checker passport2, just like the x-rite color checker classic. It seems that I need to rewrite many codes, do you have a simple way to do that? Please inform me if you can not see my uploaded image.
![DSC03354](https://user-images.githubusercontent.com/50072588/121230518-4fa38e80-c8ca-11eb-8cc6-1e0acedae111.JPG)

**Comment by KelSolaar** [comment, marked as answer]
Hi @YiteLiu,

Did you try? The ColorChecker Passport should work but it is typically expected to be fully visible.

Cheers,

Thomas

**Reply by KelSolaar** [reply]
Hi,

I just loaded your image in the example notebook and it works:

![image](https://user-images.githubusercontent.com/99779/121768857-9805c980-cbb4-11eb-84a2-b8e1322e758e.png)
![image](https://user-images.githubusercontent.com/99779/121768860-9fc56e00-cbb4-11eb-85d2-459a70c5d3d2.png)
![image](https://user-images.githubusercontent.com/99779/121768863-a7851280-cbb4-11eb-895d-e227e8a7fac0.png)
![image](https://user-images.githubusercontent.com/99779/121768867-ace25d00-cbb4-11eb-8e79-394ca52a13fc.png)

Cheers,

Thomas

**Reply by YiteLiu** [reply]
Sorry that I reply so late. Maybe there is something wrong with my environment. So glad that your results are well. That encouraged me to try again and learn python more. 

Here is an error from my traceback.
 Traceback (most recent call last):

  File "C:\Users\19379\AppData\Roaming\Python\Python38\site-packages\colour_checker_detection\detection\segmentation.py", line 461, in crop_and_level_image_with_rectangle
    M_r = cv2.getRotationMatrix2D(centroid, angle, 1)
TypeError: Can't parse 'center'. Sequence item with index 0 has a wrong type

I will reset my environment as the example code shown.
Hope I succuss this time.

**Reply by YiteLiu** [reply]
This is the first time I ask a question on Github. Thanks for your kind response again.

**Reply by KelSolaar** [reply]
What version of OpenCV are you using? The latest is having some issues with the Pypi version of colour-checker-detection. I have fixed those in the develop branch.
